sensor: rm3100: Add missing cycle_count in edata header.

anthony289478 opened the pull request:
Fixes missing cycle_count configuration in `edata->header.cycle_count` which would result in decoding of stream data to always use the 600hz scale factor and values off by a factor of 2.

This was configured when performing one-shot submissions but missed for stream submissions.

Also corrects the units in the comments I found while searching for this problem.

anthony289478 committed:
sensor: rm3100: Correct LSB/uT scale factor comment in rm3100_decoder.c

The scale factor when used as a divider from the datasheet specifies
units in LSB/uT not uT/LSB. This change corrects the comment to reflect
the correct units for the divider.

Signed-off-by: Anthony Williams <anthony289478@gmail.com>

ubieda commented on drivers/sensor/pni/rm3100/rm3100_stream.c:
```diff
@@ -29,6 +30,12 @@ static void rm3100_complete_result(struct rtio *ctx, const struct rtio_sqe *sqe,
+		edata->header.cycle_count = RM3100_CYCLE_COUNT_HIGH_ODR;
+	} else {
+		edata->header.cycle_count = RM3100_CYCLE_COUNT_DEFAULT;
+	}
```
I see this was present in one-shot mode through `rm3100_encode`. I suggest to use in streaming as well.

anthony289478 committed:
sensor: rm3100: Use rm3100_encode to populate cycle_count during stream.

Use rm3100_encode() to populate cycle_count info when
configured in streaming mode.

Previously when configured for streaming, cycle_count would
be 0 and rm3100_convert_raw_to_q31 would default to always
use the 600hz divider resulting in values off by a factor of 2.

Signed-off-by: Anthony Williams <anthony289478@gmail.com>

anthony289478 committed:
sensor: rm3100: Remove duplicate edata header encoding when streaming

Remove the manual encoding of edata->header.channels and
edata->header.timestamp as these are now handled during rm3100_encode()

Signed-off-by: Anthony Williams <anthony289478@gmail.com>
